packages/astro/src/@types/astro.ts

```diff
@@ -2408,6 +2408,133 @@ export interface AstroUserConfig {
 		 * For a complete overview and the full API reference, see [the Content Layer API RFC](https://github.com/withastro/roadmap/blob/content-layer/proposals/0047-content-layer.md) and [share your feedback](https://github.com/withastro/roadmap/pull/982).
 		 */
 		contentLayer?: boolean;
+
+		/**
+		 * @docs
+		 * @name experimental.tsconfig
+		 * @type {object}
+		 * @default `undefined`
+		 * @version 4.15.0
+		 * @description
+		 *
+		 * Enables the generation of `.astro/tsconfig.json`. This allows to exclude the `outDir` by default to avoid `astro check` false positives.
+		 *
+		 * If you enable this option, you'll need to update your `tsconfig.json` with a new `extends` value:
+		 *
+		 * ```json title="tsconfig.json" del={2} ins={3}
+		 * {
+		 *   extends: 'astro/tsconfigs/base',
+		 * 	 extends: ['astro/tsconfigs/base', './.astro/tsconfig.json']
+		 * }
+		 * ```
+		 *
+		 * If you have `include` or `exclude` inside your root `tsconfig.json`, you'll need to move them to your Astro config:
+		 *
+		 * ```json title="tsconfig.json" del={2-3}
+		 * {
+		 *   include: ['foo'],
+		 *   exclude: ['bar']
```

Comment:
Can we use something real here instead of foo and bar? Seeing a real example helps people know: Are these file paths? If so, absolute/relative? Are these expressions?

Reply:
```suggestion
		 *   include: ['src/**/*'],
		 *   exclude: ['node_modules']
```

I linked the official docs on Discord but I'm going to link them here as well: https://www.typescriptlang.org/tsconfig/#include. So those are relative to the project root

Reply:
Love this change, perfect!

Reply:
I think it still makes sense yes, if you wanted not to check root ts files like the astro config